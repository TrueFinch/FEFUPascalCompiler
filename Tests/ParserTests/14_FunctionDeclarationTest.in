program a;

function add(const a, b: integer; out res: integer) : integer;
forward;

function add(const a, b: integer; out res: integer) : integer;
begin
    res := a + b;
end;

begin
end.