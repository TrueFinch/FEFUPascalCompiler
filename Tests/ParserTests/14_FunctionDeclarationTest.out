                                                                                    a                                                                                     
                                                                                    ┼                                                                                     
                                                                                MainBlock                                                                                 
                                  ┌─────────────────────────────────────────────────┴────────────────────────┬─────────────────────────────────────────────────┐          
                               FuncDecl                                                                   FuncDecl                                     CompoundStatement  
                              ┌───┴──────────────────────────────┐                                  ┌────────┴──────────────────────────────┐                             
                         FuncHeader                           forward                          FuncHeader                            SubroutineBlock                      
  ┌──────────────┬────────────┴─────────┬───────────────┐               ┌──────────────┬────────────┴─────────┬───────────────┐             ┼                             
 add    FormalParamSection      FormalParamSection   integer           add    FormalParamSection      FormalParamSection   integer  CompoundStatement                     
        ┌────┬──┬┴─────┐        ┌────┬──┴────┐                                ┌────┬──┬┴─────┐        ┌────┬──┴────┐                    ┌───┴                             
      const  a  b  SimpleType  out  res  SimpleType                         const  a  b  SimpleType  out  res  SimpleType              :=                                 
                      ┌┴                    ┌┴                                              ┌┴                    ┌┴                 ┌──┴─┐                               
                   integer               integer                                         integer               integer              res   +                               
                                                                                                                                         ┌┴─┐                             
                                                                                                                                         a  b                             